parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -31,39 +35,75 @@
  */
 public abstract class Statistics<T extends Comparable<T>> {
 
+  private final PrimitiveComparator<T> comparator;
   private boolean hasNonNullValue;
   private long num_nulls;
 
-  public Statistics() {
+  Statistics(PrimitiveComparator<T> comparator) {
     hasNonNullValue = false;
     num_nulls = 0;
+    this.comparator = comparator;
   }
 
   /**
    * Returns the typed statistics object based on the passed type parameter
-   * @param type PrimitiveTypeName type of the column
+   * 
+   * @param type
+   *          PrimitiveTypeName type of the column
    * @return instance of a typed statistics class
+   * @deprecated Use {@link #createStats(Type)} instead
    */
+  @Deprecated
   public static Statistics getStatsBasedOnType(PrimitiveTypeName type) {
     switch(type) {
-    case INT32:
-      return new IntStatistics();
-    case INT64:
-      return new LongStatistics();
-    case FLOAT:
-      return new FloatStatistics();
-    case DOUBLE:
-      return new DoubleStatistics();
-    case BOOLEAN:
-      return new BooleanStatistics();
-    case BINARY:
-      return new BinaryStatistics();
-    case INT96:
-      return new BinaryStatistics();
-    case FIXED_LEN_BYTE_ARRAY:
-      return new BinaryStatistics();
-    default:
-      throw new UnknownColumnTypeException(type);
+      case INT32:
```

Comment:
Nit: I think the only change here is indentation, but below there are switch statements that align the indentation of `case` with `switch`. If you're going to make this change, please be consistent with indentation in switch statements elsewhere.

Reply:
The formatting of switch-case is already mixed in the code base. Do we have formatter rules for IDEs like Eclipse somewhere? I think, it would be much easier to have the same format everywhere.

Reply:
For the most part, we use the defaults in IntelliJ, which I believe follows google's style: https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation

According to that, case should be indented 2 spaces from switch, so lets go with that for case statements affected by this patch.

Reply:
Looks like you went with the opposite, where case is not indented from switch.

Reply:
Yep, sorry. Will correct it.